src/components/microlearning/data/utils.js

```diff
@@ -0,0 +1,31 @@
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+
+dayjs.extend(duration);
+
+const formatDuration = (durationInSeconds) => {
+  const time = dayjs.duration({ seconds: durationInSeconds });
+  const minutes = Math.floor(time.asMinutes());
+  const seconds = (durationInSeconds % 60).toFixed(0).padStart(2, '0');
+  return `${minutes}:${seconds}`;
+};
+
+const formatSkills = (skills) => skills['list-item'].map(skill => ({
+  name: skill.name,
+  description: skill.description,
+  category: skill.category.name,
+  subcategory: skill.subcategory.name,
+}));
+
+export const transformVideoData = (data) => {
+  const transformedData = {
```

Comment:
nit: It would be better to skip variable definition here and simply return the new object.

```
export const transformVideoData = (data) => ({
    videoUrl: data?.json_metadata?.download_link,
    courseTitle: data?.parent_content_metadata?.title,
    videoSummary: data?.summary_transcripts['list-item'],
    transcriptUrls: data?.json_metadata?.transcript_urls,
    videoSkills: formatSkills(data?.skills),
    videoDuration: formatDuration(data?.json_metadata?.duration),
});
```